dashboard/client/src/components/NewEventTable.tsx

```diff
@@ -0,0 +1,371 @@
+  },
+  message: {
+    maxWidth: "200",
+  },
+  pagination: {
+    marginTop: theme.spacing(3),
+  },
+}));
+
+const SOURCE_TYPE_OPTIONS = [
+  "common",
+  "core_worker",
+  "gcs",
+  "raylet",
+  "jobs",
+  "serve",
+  "cluster_lifecycle",
+  "autoscaler",
+];
+
+const SEVERITY_LEVEL_OPTIONS = ["info", "debug", "warning", "error", "tracing"];
+
+const COLUMNS = [
+  { label: "Severity", align: "center" },
+  { label: "Timestamp", align: "center" },
+  { label: "Source", align: "center" },
+  { label: "Custom Fields", align: "left" },
+  { label: "Message", align: "left" },
+];
+type Align = "inherit" | "left" | "center" | "right" | "justify";
+
+type Filters = {
+  sourceType: string[]; // TODO: Chao, multi-select severity level in filters button is a P1
+  severityLevel: string[]; // TODO: Chao, multi-select severity level in filters button is a P1
+  entityName: string | undefined;
+  entityId: string | undefined;
+};
+
+const useEventTable = (props: EventTableProps) => {
+  const { defaultSeverityLevels, entityName, entityId } = props;
+  const [loading, setLoading] = useState(true);
+  const { changeFilter: _changeFilter, filterFunc } = useFilter();
+  const [filters, _setFilters] = useState<Filters>({
+    sourceType: [],
+    severityLevel: defaultSeverityLevels || [],
+    entityName, // We used two fields(entityName, entityId) because we will support select entityName by dropdown and input entityId by TextField in the future.
+    entityId, // id or *
+  });
+
+  const [events, setEvents] = useState<Event[]>([]);
+
+  const [pagination, setPagination] = useState({
+    pageNo: 1,
+    pageSize: 10,
+    total: 0,
+  });
+  const { pageSize } = pagination;
+
+  const changePage = (key: string, value: number) => {
+    setPagination({ ...pagination, [key]: value });
+  };
+
+  const changeFilter: typeof _changeFilter = (...params) => {
+    _changeFilter(...params);
+    setPagination({
+      ...pagination,
+      pageNo: 1,
+    });
+  };
+
+  const setFilters: typeof _setFilters = (...params) => {
+    _setFilters(...params);
+    setPagination({
+      ...pagination,
+      pageNo: 1,
+    });
+  };
+
+  useEffect(() => {
+    const getEvent = async () => {
+      try {
+        const params = transformFiltersToParams(filters);
+        const rsp = await getEvents(params); // We don't useSWR since we need to get real time events data once filters changed
+        const events = rsp?.data?.data?.result?.result;
+        if (events) {
+          setEvents(events); // We sort the event by timestamp in the backend
+        }
+      } catch (e) {
+        console.error("getEvent error: ", e);
+      } finally {
+        setLoading(false);
+      }
+    };
+    getEvent();
+  }, [filters]);
+
+  const realLen = events.filter(filterFunc).length;
+  useEffect(() => {
+    setPagination((p) => ({
+      ...p,
+      total: Math.ceil(realLen / p.pageSize),
+      pageNo: 1,
+    }));
+  }, [realLen, pageSize]);
```

Comment:
what is the purpose of this effect? Why can't pagination be manipulated directly?

Reply:
I got your point, since we don't support change pageSize, we could remove this useEffect